### PR TITLE
Task 1: implemented Inventory Create DAO and testing

### PR DESCRIPTION
**Pull request by KRB065**

### CODE REVIEW

File: starter/src/test/java/com/starter/fullstack/dao/InventoryDAOTest.java [hunk 1]
```diff
@@ -36,7 +37,17 @@ public void setup() {
...
+  // Test the create function. Initialize inventory and then add it to mongo collection with the create function. Print the ID before and after to check for null
+  @Test
+  public void createTest() {
+    Inventory inventory1 = new Inventory();
+    inventory1.setName(NAME);
+    inventory1.setId("877241");
+    inventory1.setProductType(PRODUCT_TYPE);
+    System.out.print("The initial ID is: " + inventory1.getId());
+    this.inventoryDAO.create(inventory1);
+    System.out.println("After Create Method ID: " + inventory1.getId());
```
Comment (ohnoitsyou) [inline starter/src/test/java/com/starter/fullstack/dao/InventoryDAOTest.java:49]: Part of testing is also verifying that after you run the method under test, `create` in this case, by using `assert`s

File: starter/src/test/java/com/starter/fullstack/dao/InventoryDAOTest.java [hunk 2]
```diff
@@ -36,7 +37,17 @@ public void setup() {
   public void tearDown() {
     this.mongoTemplate.dropCollection(Inventory.class);
   }
-
+  // Test the create function. Initialize inventory and then add it to mongo collection with the create function. Print the ID before and after to check for null
+  @Test
+  public void createTest() {
+    Inventory inventory1 = new Inventory();
+    inventory1.setName(NAME);
+    inventory1.setId("877241");
```
Comment (ohnoitsyou) [inline starter/src/test/java/com/starter/fullstack/dao/InventoryDAOTest.java:45]: Try using the ID that's defined as a constant from up above

**Review by ohnoitsyou — CHANGES_REQUESTED** [review]

**Review by KRB065 — COMMENTED** [review]
Added assert statement

File: starter/target/surefire-reports/com.starter.fullstack.dao.InventoryDAOTest.txt [hunk 3]
```diff
@@ -0,0 +1,8 @@
+-------------------------------------------------------------------------------
+Test set: com.starter.fullstack.dao.InventoryDAOTest
+-------------------------------------------------------------------------------
+Tests run: 2, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 0.453 s <<< FAILURE! - in com.starter.fullstack.dao.InventoryDAOTest
+createTest  Time elapsed: 0.046 s  <<< FAILURE!
+java.lang.AssertionError: Values should be different. Actual: 62db713a868abc3a666887d7
+	at com.starter.fullstack.dao.InventoryDAOTest.createTest(InventoryDAOTest.java:50)
+
```
Comment (ohnoitsyou) [inline starter/target/surefire-reports/com.starter.fullstack.dao.InventoryDAOTest.txt]: Take a look at this test and see why it's failing.

You should also not include the test report in the repository.

File: starter/target/surefire-reports/com.starter.fullstack.dao.ProductDAOTest.txt [hunk 4]
```diff
@@ -0,0 +1,4 @@
+-------------------------------------------------------------------------------
+Test set: com.starter.fullstack.dao.ProductDAOTest
+-------------------------------------------------------------------------------
+Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.349 s - in com.starter.fullstack.dao.ProductDAOTest
```
Comment (ohnoitsyou) [inline starter/target/surefire-reports/com.starter.fullstack.dao.ProductDAOTest.txt]: Please remove the test report.

File: starter/target/surefire-reports/com.starter.fullstack.rest.ProductControllerTest.txt [hunk 5]
```diff
@@ -0,0 +1,4 @@
+-------------------------------------------------------------------------------
+Test set: com.starter.fullstack.rest.ProductControllerTest
+-------------------------------------------------------------------------------
+Tests run: 3, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 13.475 s - in com.starter.fullstack.rest.ProductControllerTest
```
Comment (ohnoitsyou) [inline starter/target/surefire-reports/com.starter.fullstack.rest.ProductControllerTest.txt]: Please remove the test report.

File: starter/src/test/java/com/starter/fullstack/dao/InventoryDAOTest.java [hunk 6]
```diff
@@ -47,6 +47,7 @@ public void createTest() {
     System.out.print("The initial ID is: " + inventory1.getId());
     this.inventoryDAO.create(inventory1);
     System.out.println("After Create Method ID: " + inventory1.getId());
+    Assert.assertTrue(inventory1.getId() != "877241");
```
Comment (ohnoitsyou) [inline starter/src/test/java/com/starter/fullstack/dao/InventoryDAOTest.java:50]: This is part of the reason you should use the variables you have defined at the class level. Although the contents of the strings are the same, they are technically two different strings. You should use a variable in both the creation of the item and the checking of the item.

File: starter/.gitignore [hunk 7]
```diff
@@ -0,0 +1,10 @@
+*.html
+*.class
+*.xml
+*.gif
+*.properties
+*.class
+*.jar
+*.class
+*.lst
+*.jar
```
Comment (ohnoitsyou) [inline starter/.gitignore]: Although not a huge deal, you have some duplicates.

**Review by ohnoitsyou — CHANGES_REQUESTED** [review]

**Pull request closed without merging**